src/core/thread/mle_router.cpp

```diff
@@ -1774,6 +1777,40 @@ void MleRouter::HandleStateUpdateTimer(void)
         break;
     }
 
+    if (mIsRouterRestoringChildren)
```

Comment:
I agree that we should not limit to one message per second.  However, I don't think we need to have a separate timer.  Instead, let's try to buffer/send as messages as we can.  If we run out of buffer, then we can try to continue on the next call to `HandleStateUpdateTimer()`